File: graphview/src/main/java/de/blox/graphview/layered/SugiyamaNodeData.java
package de.blox.graphview.layered;

import java.util.HashSet;
import java.util.Set;

import de.blox.graphview.Node;

/**
 * Class to save additional data used by the sugiyama algorithm.
 */
class SugiyamaNodeData {
    Set<Node> reversed = new HashSet<>();
    boolean dummy = false;
    int median = -1;
    int layer = -1;

    public boolean isDummy() {
        return dummy;
    }

    public boolean isReversed() {
        return !reversed.isEmpty();
    }

    @Override
    public String toString() {
        return "SugiyamaNodeData{" +
                ", reversed=" + reversed +
                ", dummy=" + dummy +
                ", median=" + median +
                ", layer=" + layer +
                '}';
    }
}

File: graphview/src/main/java/de/blox/graphview/layered/SugiyamaNodeData.kt
package de.blox.graphview.layered

import de.blox.graphview.Node

internal class SugiyamaNodeData {
    val reversed = mutableSetOf<Node>()
    var isDummy = false
    var median = -1
    var layer = -1

    val isReversed: Boolean
        get() = reversed.isNotEmpty()

    override fun toString(): String {
        return "SugiyamaNodeData{" +
                ", reversed=" + reversed +
                ", dummy=" + isDummy +
                ", median=" + median +
                ", layer=" + layer +
                '}'.toString()
    }
}
